client/my-sites/sidebar/sidebar.jsx

```diff
@@ -326,7 +326,15 @@ module.exports = React.createClass( {
 			linkClass += ' is-paid-plan';
 		}
 
-		let planName = site.plan.product_name_short;
+		let planName = site.plan.product_name_short,
+			labelClass;
+
+		if ( abtest( 'plansUpgradeButton' ) === 'button' && productsValues.isFreePlan( site.plan ) ) {
+			labelClass = 'add-new';
+			planName = 'Upgrade'; // TODO: translate this string if the test is removed
+		} else {
+			labelClass = 'plan-name';
+		}
```

Comment:
I'd like to suggest the following approach which should be easier to understand - since having another block of code with a very similar condition (`isFreePlan` versus `isFreeTrial`) that also updates the plan name is a bit misleading in my opinion):

```javascript
let planName = site.plan.product_name_short,
  labelClass = 'plan-name';

if ( (abtest( 'plansUpgradeButton' ) === 'button') && productsValues.isFreePlan( site.plan ) ) {
  labelClass = 'add-new';
  planName = 'Upgrade'; // TODO: translate this string if the test is removed
} else if ( productsValues.isFreeTrial( site.plan ) ) {
  planName = this.translate( 'Trial' );
}
```